apps/src/codebridge/utils/fileUtils.ts

```diff
@@ -0,0 +1,27 @@
+import {START_SOURCES} from '@cdo/apps/lab2/constants';
+import {getAppOptionsEditBlocks} from '@cdo/apps/lab2/projects/utils';
+
+import {ProjectFile} from '../types';
+
+export function shouldShowFile(file: ProjectFile) {
+  const isStartMode = getAppOptionsEditBlocks() === START_SOURCES;
+
+  // If we are in start mode, show all files. If we are not in start mode,
+  // hide files that are hidden or are validation.
+  return isStartMode ? true : !file.hidden && !file.validation;
+}
+
+export function getFileIcon(file: ProjectFile) {
+  const isStartMode = getAppOptionsEditBlocks() === START_SOURCES;
+  let icon = 'fa-solid ';
+  if (isStartMode) {
+    icon += file.validation
+      ? 'fa-flask'
+      : file.hidden
+      ? 'fa-eye-slash'
+      : 'fa-eye';
```

Comment:
I sometimes wonder if nested ternaries like this are harder to read than `if...else if...else`. No need to change, but thought I'd flag in case you or anyone else has an opinion.

Reply:
I definitely find nested ternaries harder to read than if/else ladders, but considering how many of them we have in the codebase I'm also not gonna demand it be replaced.

Reply:
I'm happy to change it!